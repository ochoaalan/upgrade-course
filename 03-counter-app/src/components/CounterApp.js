// import React, { Fragment } from 'react';
import React, { useState } from "react";
import PropTypes from "prop-types";

function CounterApp({ value = 10, subtitulo }) {
  const [counter, setCounter] = useState( value );

  // handleAdd
  const handleAdd = (e) => {
    setCounter( counter + 1 );
    // otra forma de usar setCounter
    // setCounter((contador) => contador + 1);
  };

    // handleSubtract
    const handleSubtract = (e) => {
      setCounter( counter - 1 );
    };

    // handleReset
    const handleReset = (e) => {
      setCounter( value );
    }

  return (
    <>
      <h1>Counter App</h1>
      <p>{ subtitulo }</p>
      <h2> {counter} </h2>

      {/* <button onClick={(e) => { handleAdd(e) }}>+1</button> */}
      {/* simplificando funcion */}
      <button onClick={ handleAdd }>+1</button>
      <button onClick={ handleReset }>Reset</button>
      <button onClick={ handleSubtract }>-1</button>
      {/* otra forma usando la funcion setcounter aqui directo */}
      {/* <button onClick={ ()=> setCounter( value ) }>Reset</button>
      <button onClick={ ()=> setCounter( counter -1) }>-1</button> */}
    </>
  );
}

  CounterApp.propTypes = {
// value: PropTypes.number.isRequired,
  value: PropTypes.number
  };

  CounterApp.defaultProps = {
  subtitulo: 'Soy un subtitulo'
  }

export default CounterApp;
