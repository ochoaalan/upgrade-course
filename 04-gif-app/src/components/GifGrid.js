import React from 'react';
import PropTypes from 'prop-types';

import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({ category }) => {

  const { data:images, loading } = useFetchGifs( category ); // data:images  cambia el nombre de data a images en la desestrutcturacion para que tenga mas sentido nuestro codigo

  return (
    <>
      <h3 className='animate__animated animate__fadeIn'>{ category }</h3>

      {loading && <p className='animate__animated animate__flash'>Loading...</p>}

        <div className='card-grid'>
        {
          images.map( img => ( 
            <GifGridItem 
              key={ img.id }
              {...img }
            />
          ))
        }
        </div>
    </>

  )
}

GifGrid.propTypes = {
  category: PropTypes.string.isRequired
}