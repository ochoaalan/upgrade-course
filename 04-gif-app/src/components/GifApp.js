import React, { useState } from 'react';
import { AddCategory } from './AddCategory';
import { GifGrid } from './GifGrid';

export const GifApp = ({ defaultCategories = [] }) => {

// const [categories, setCategories] = useState(['Castlevania']);
const [categories, setCategories] = useState( defaultCategories );

  return (
    <div>
      <h2>GifApp</h2>
      <AddCategory setCategories={setCategories} />
      <hr />

      <ol>
        { 
          categories.map( category => (
            <GifGrid 
              key={ category }
              category={ category }
            />
          ))
        }
      </ol>
    </div>
  )
}
