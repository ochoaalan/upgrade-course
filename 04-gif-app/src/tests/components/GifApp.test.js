import { React } from "react";
import { shallow } from 'enzyme';
import { GifApp } from '../../components/GifApp';

describe('Pruebas en <GifApp', () => {

  test('Debe mostrse correctamente', () => {
    
    const wrapper = shallow( <GifApp /> );
    expect( wrapper ).toMatchSnapshot();

  });

  test('Debe de mostrar una lista de categorias ', () => {
    
    const categories = ['Megaman', 'Gears of War'];
    const wrapper = shallow( <GifApp defaultCategories={ categories } /> );

    expect( wrapper ).toMatchSnapshot();
    expect( wrapper .find('GifGrid').length ).toBe( categories.length );

  })
  
  
  
})
