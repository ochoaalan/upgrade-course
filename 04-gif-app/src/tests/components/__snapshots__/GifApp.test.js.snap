// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifApp Debe de mostrar una lista de categorias  1`] = `
<div>
  <h2>
    GifApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Megaman"
      key="Megaman"
    />
    <GifGrid
      category="Gears of War"
      key="Gears of War"
    />
  </ol>
</div>
`;

exports[`Pruebas en <GifApp Debe mostrse correctamente 1`] = `
<div>
  <h2>
    GifApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</div>
`;
